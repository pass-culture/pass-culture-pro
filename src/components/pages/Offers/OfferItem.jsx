import { Icon, pluralize, requestData } from 'pass-culture-shared'
import classnames from 'classnames'
import get from 'lodash.get'
import moment from 'moment'
import React, { Component } from 'react'
import Dotdotdot from 'react-dotdotdot'
import { connect } from 'react-redux'
import { withRouter } from 'react-router'
import { NavLink } from 'react-router-dom'
import { compose } from 'redux'

import Price from '../../layout/Price'
import Thumb from '../../layout/Thumb'
import aggregatedStockSelector from '../../../selectors/aggregatedStock'
import eventSelector from '../../../selectors/event'
import maxDateSelector from '../../../selectors/maxDate'
import mediationsSelector from '../../../selectors/mediations'
import eventOccurrencesSelector from '../../../selectors/eventOccurrences'
import stocksSelector from '../../../selectors/stocks'
import thingSelector from '../../../selectors/thing'
import thumbUrlSelector from '../../../selectors/thumbUrl'
import typeSelector from '../../../selectors/type'
import venueSelector from '../../../selectors/venue'
import offerrerSelector from '../../../selectors/offerer'
import { offerNormalizer } from '../../../utils/normalizers'

class OccasionItem extends Component {
  onDeactivateClick = event => {
    const { dispatch, offer } = this.props
    const { id, isActive } = offer || {}
    dispatch(
      requestData('PATCH', `offers/${id}`, {
        body: {
          isActive: !isActive,
        },
        normalizer: offerNormalizer,
        isMergingDatum: true,
        isMutatingDatum: true,
        isMutaginArray: false,
      })
    )
  }

  render() {
    const {
      aggregatedStock,
      event,
      eventOccurrences,
      location: { search },
      maxDate,
      mediations,
      offer,
      stocks,
      thing,
      thumbUrl,
      type,
      offerrer,
      venue,
    } = this.props
    const { isNew } = offer || {}
    const { available, groupSizeMin, groupSizeMax, priceMin, priceMax } =
      aggregatedStock || {}
    const { name, createdAt } = event || thing || {}

    const mediationsLength = get(mediations, 'length')

    return (
      <li
        className={classnames('offer-item', {
          active: offer.isActive,
          event,
          thing,
        })}>
        <Thumb alt="offre" src={thumbUrl} />
        <div className="list-content">
          <NavLink
            className="name"
            to={`/offres/${offer.id}${search}`}
            title={name}>
            <Dotdotdot clamp={1}>{name}</Dotdotdot>
          </NavLink>
          <ul className="infos">
            <li className="is-uppercase">
              {get(type, 'label') || (event ? 'Evénement' : 'Objet')}
            </li>
            <li>
              <span className="label">Structure :</span>
              {offerrer && offerrer.name}
            </li>
            <li>
              <span className="label">Lieu :</span> {venue && venue.name}
            </li>
          </ul>
          <ul className="infos">
            {isNew && (
              <li>
                <div className="recently-added" />
              </li>
            )}
            {false && moment(createdAt).isAfter(moment().add(-1, 'days')) && (
              <li>
                <div className="recently-added" />
              </li>
            )}
            <li
              title={
                groupSizeMin > 0
                  ? groupSizeMin === groupSizeMax
                    ? `minimum ${pluralize(groupSizeMin, 'personnes')}`
                    : `entre ${groupSizeMin} et ${groupSizeMax} personnes`
                  : undefined
              }>
              {
                // DISABLE GROUP SIZE AMBIGUITY FOR THE MOMENT
                /*groupSizeMin === 0 && (
                <div>
                  <Icon svg="picto-user" /> {'ou '} <Icon svg="picto-group" />
                </div>
                )*/
              }
              {groupSizeMin === 1 && <Icon svg="picto-user" />}
              {groupSizeMin > 1 && (
                <div>
                  <Icon svg="picto-group" />,{' '}
                  <p>
                    {groupSizeMin === groupSizeMax
                      ? groupSizeMin
                      : `${groupSizeMin} - ${groupSizeMax}`}
                  </p>
                </div>
              )}
            </li>
            <li>
              <NavLink
                className="has-text-primary"
                to={`/offres/${offer.id}?gestion`}>
                {event
                  ? pluralize(get(eventOccurrences, 'length'), 'dates')
                  : get(stocks, 'length') + ' prix'}
              </NavLink>
            </li>
            <li>{maxDate && `jusqu'au ${maxDate.format('DD/MM/YYYY')}`}</li>
            <li>{available ? `encore ${available} places` : '0 place'} </li>
            <li>
              {priceMin === priceMax ? (
                <Price value={priceMin || 0} />
              ) : (
                <span>
                  <Price value={priceMin} /> - <Price value={priceMax} />
                </span>
              )}
            </li>
          </ul>
          <ul className="actions">
            <li>
              <NavLink
                to={`/offres/${offer.id}${
                  mediationsLength ? '' : `/accroches/nouveau${search}`
                }`}
                className={`button addMediations is-small ${
                  mediationsLength ? 'is-secondary' : 'is-primary is-outlined'
                }`}>
                <span className="icon">
                  <Icon svg="ico-stars" />
                </span>
                <span>
                  {get(mediations, 'length')
                    ? 'Accroches'
                    : 'Ajouter une Accroche'}
                </span>
              </NavLink>
            </li>
            <li>
              <NavLink
                to={`/offres/${offer.id}`}
                className="button is-secondary is-small edit-link">
                <Icon svg="ico-pen-r" />
                Modifier
              </NavLink>

              <button
                className="button is-secondary is-small activ-switch"
                onClick={this.onDeactivateClick}>
                {offer.isActive ? (
                  <span>
                    <Icon svg="ico-close-r" />
                    Désactiver
                  </span>
                ) : (
                  'Activer'
                )}
              </button>
            </li>
          </ul>
        </div>
      </li>
    )
  }
}

OccasionItem.defaultProps = {
  maxDescriptionLength: 300,
}

function mapStateToProps(state, ownProps) {
  const { id, eventId, thingId } = ownProps.offer
  const event = eventSelector(state, eventId)
  const thing = thingSelector(state, thingId)
  const typeValue = get(event, 'type') || get(thing, 'type')
  const eventOccurrences = eventOccurrencesSelector(state, id)
  const venue = venueSelector(state, ownProps.offer.venueId)
  const offerrer = offerrerSelector(state, venue.managingOffererId)
  return {
    aggregatedStock: aggregatedStockSelector(
      state,
      id,
      event && eventOccurrences
    ),
    event,
    eventOccurrences,
    maxDate: maxDateSelector(state, id),
    mediations: mediationsSelector(state, id),
    stocks: stocksSelector(state, id, event && eventOccurrences),
    thing,
    thumbUrl: thumbUrlSelector(state, id, eventId, thingId),
    type: typeSelector(state, typeValue),
    venue,
    offerrer,
  }
}

export default compose(
  withRouter,
  connect(mapStateToProps)
)(OccasionItem)
