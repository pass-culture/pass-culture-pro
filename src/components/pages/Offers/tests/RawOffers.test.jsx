import React from 'react'
import { shallow } from 'enzyme'
import { NavLink } from 'react-router-dom'

import RawOffers from '../RawOffers'
import OfferItem from '../OfferItem'
import mockedOffers from './offersMock'

const dispatchMock = jest.fn()
const queryChangeMock = jest.fn()
const initialProps = {
  dispatch: dispatchMock,
  offers: [],
  location: {
    pathname: '/offres',
  },
  pagination: {
    apiQuery: {
      keywords: null,
      offererId: null,
      orderBy: 'offer.id+desc',
      venueId: null,
    },
  },
  query: {
    change: queryChangeMock,
    parse: () => ({ orderBy: 'offer.id desc' }),
  },
  search: '',
  types: [],
  user: {},
}

describe('src | components | pages | Offers | RawOffers', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      // when
      const wrapper = shallow(<RawOffers {...initialProps} />)

      // then
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
      dispatchMock.mockClear()
      queryChangeMock.mockClear()
    })
  })

  describe('render', () => {
    describe('when arriving on index page', () => {
      it('should list all offers', () => {
        // given
        const expected = {
          hasMore: false,
          isLoading: true,
          keywordsValue: undefined,
        }

        // when
        const wrapper = shallow(<RawOffers {...initialProps} />)

        // then
        expect(wrapper.state()).toEqual(expected)
        dispatchMock.mockClear()
        queryChangeMock.mockClear()
      })
    })
    describe('OfferItem', () => {
      it('should render items corresponding to offers', () => {
        // given
        initialProps.offers = mockedOffers

        // when
        const wrapper = shallow(<RawOffers {...initialProps} />)
        const offerItem = wrapper.find(OfferItem)

        // then
        expect(offerItem).toHaveLength(mockedOffers.length)
      })
    })
    describe('When offers showned for an offerer or a venue', () => {
      describe('NavLink to create offer', () => {
        describe('When user isAdmin', () => {
          it('should display NavLink', () => {
            // given
            initialProps.user = {
              isAdmin: true,
            }

            // when
            const wrapper = shallow(<RawOffers {...initialProps} />)
            const navLink = wrapper.find(NavLink)

            // then
            expect(navLink).toHaveLength(0)
          })
        })
        describe('When structure (or offererId)', () => {
          it('should render link properly', () => {
            // given
            initialProps.user = {
              isAdmin: false,
            }
            const parseMock = () => ({ structure: 'TEST' })
            initialProps.query.parse = parseMock

            // when
            const wrapper = shallow(<RawOffers {...initialProps} />)

            const navLink = wrapper.find(NavLink)

            // then
            expect(navLink).toHaveLength(1)
            expect(navLink.props().to).toEqual('/offres/nouveau?structure=TEST')
          })
        })
        describe('When lieu or (VenueId)', () => {
          it('should render link properly', () => {
            // given
            initialProps.user = {
              isAdmin: false,
            }
            const parseMock = () => ({ lieu: 'LIEU' })
            initialProps.query.parse = parseMock

            // when
            const wrapper = shallow(<RawOffers {...initialProps} />)

            const navLink = wrapper.find(NavLink)

            // then
            expect(navLink).toHaveLength(1)
            expect(navLink.props().to).toEqual('/offres/nouveau?lieu=LIEU')
          })
        })
      })
    })
  })

  describe('functions', () => {
    describe('constructor', () => {
      it('should dispatch assignData when component is constructed', () => {
        // when
        shallow(<RawOffers {...initialProps} />)
        const expectedAssignData = {
          patch: {
            offers: [],
          },
          type: 'ASSIGN_DATA',
        }

        // then
        expect(dispatchMock.mock.calls[0][0]).toEqual(expectedAssignData)
        dispatchMock.mockClear()
        queryChangeMock.mockClear()
      })
    })

    describe('componentDidMount', () => {
      describe('when there is pagination', () => {
        it.skip('should call query change with good params', () => {
          // given
          const parseMock = () => ({ page: 1 })
          initialProps.query.parse = parseMock
          initialProps.query.change = queryChangeMock

          // when
          const wrapper = shallow(<RawOffers {...initialProps} />)

          wrapper.instance().componentDidMount()

          // then
          expect(queryChangeMock.mock.call).toEqual({ page: null })
          queryChangeMock.mockClear()
        })
      })

      describe('when there is no pagination', () => {
        it('should dispatch handle RequestData', () => {
          // given
          const parseMock = () => ({
            keywords: 'océan',
            orderBy: 'offer.id desc',
          })
          initialProps.query.parse = parseMock

          // when
          const wrapper = shallow(<RawOffers {...initialProps} />)
          wrapper.instance().componentDidMount()
          const expected = {
            config: {},
            method: 'GET',
            path: 'types',
            type: 'REQUEST_DATA_GET_TYPES',
          }

          // then
          expect(dispatchMock.mock.calls[1][0]).toEqual(expected)
          dispatchMock.mockClear()
        })
      })
    })

    describe('componentDidUpdate', () => {
      describe('when location has change', () => {
        it('should dispatch assignData when component is rendered', () => {
          // given
          const prevProps = initialProps

          initialProps.location = {
            pathname: '/offres',
            search: '?orderBy=offer.id+desc',
          }

          // when
          const wrapper = shallow(<RawOffers {...initialProps} />)
          wrapper.instance().componentDidUpdate(prevProps)

          const expectedAssignData = {
            config: {},
            method: 'GET',
            path: 'types',
            type: 'REQUEST_DATA_GET_TYPES',
          }

          // the
          expect(dispatchMock.mock.calls[1][0]).toEqual(expectedAssignData)

          dispatchMock.mockClear()
        })
      })
    })

    describe('onSubmit', () => {
      // when
      const wrapper = shallow(<RawOffers {...initialProps} />)
      const event = Object.assign(jest.fn(), {
        preventDefault: () => {},
        target: {
          elements: {
            search: {
              value: 'AnyWord',
            },
          },
        },
      })

      describe('when keywords is not an empty string', () => {
        it('should update state with isLoading to true', () => {
          // when
          wrapper.instance().onSubmit(event)
          const expected = {
            hasMore: false,
            isLoading: true,
            keywordsValue: undefined,
          }

          // then
          expect(wrapper.state()).toEqual(expected)
        })

        it('should dispatch assignData with good params', () => {
          // when
          wrapper.instance().onSubmit(event)

          const expected = {
            patch: {
              offers: [],
            },
            type: 'ASSIGN_DATA',
          }

          // then
          expect(dispatchMock.mock.calls[0][0]).toEqual(expected)
          dispatchMock.mockClear()
        })

        it('should change query', () => {
          // when
          wrapper.instance().onSubmit(event)
          const expected = {
            keywords: 'AnyWord',
            page: null,
          }

          // then
          expect(queryChangeMock.mock.calls[0][0]).toEqual(expected)
          queryChangeMock.mockClear()
        })
      })

      describe('when keywords is an empty string', () => {
        it('should change query with mots-clés setted to null', () => {
          // given
          const eventEmptyWord = Object.assign(jest.fn(), {
            preventDefault: () => {},
            target: {
              elements: {
                search: {
                  value: '',
                },
              },
            },
          })

          // when
          wrapper.instance().onSubmit(eventEmptyWord)
          const expected = {
            keywords: null,
            page: null,
          }

          // then
          expect(queryChangeMock.mock.calls[0][0]).toEqual(expected)
          queryChangeMock.mockClear()
        })
      })
    })
  })
})
