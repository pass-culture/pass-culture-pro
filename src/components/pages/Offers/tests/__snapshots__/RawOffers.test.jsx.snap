// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | pages | Offers | RawOffers snapshot should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RawOffers
    dispatch={[Function]}
    location={
        Object {
            "pathname": "/offres",
          }
    }
    offers={Array []}
    pagination={
        Object {
            "apiQuery": Object {
              "keywords": null,
              "offererId": null,
              "orderBy": "offer.id+desc",
              "venueId": null,
            },
          }
    }
    query={
        Object {
            "change": [Function],
            "parse": [Function],
          }
    }
    search=""
    types={Array []}
    user={Object {}}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <HeroSection
          title="Vos offres"
>
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="cta button is-primary"
                    to="/offres/nouveau"
          >
                    <span
                              className="icon"
                    >
                              <Icon
                                        svg="ico-offres-w"
                              />
                    </span>
                    <span>
                              Créer une offre
                    </span>
          </NavLink>
</HeroSection>,
        <form
          className="section"
          onSubmit={[Function]}
>
          <label
                    className="label"
          >
                    Rechercher une offre :
          </label>
          <div
                    className="field is-grouped"
          >
                    <p
                              className="control is-expanded"
                    >
                              <input
                                        className="input"
                                        defaultValue={undefined}
                                        id="search"
                                        placeholder="Saisissez un ou plusieurs mots complets"
                                        type="text"
                              />
                    </p>
                    <p
                              className="control"
                    >
                              <button
                                        className="button is-primary is-outlined"
                                        id="search-button"
                                        type="submit"
                              >
                                        OK
                              </button>
                               
                              <button
                                        className="button is-secondary"
                                        disabled={true}
                              >
                                         
                                        <Icon
                                                  svg="ico-filter"
                                        />
                                         
                              </button>
                    </p>
          </div>
</form>,
        <ul
          className="section"
/>,
        <div
          className="section"
>
          <withRouter(_withQueryRouter)
                    className="offers-list main-list"
                    element="ul"
                    hasMore={false}
                    isLoading={true}
                    loader={
                              <Spinner
                                        Tag="div"
                                        className={null}
                                        dotFrequency={500}
                                        label="Chargement"
                                        style={null}
                              />
                    }
                    useWindow={true}
          >
                    
          </withRouter(_withQueryRouter)>
</div>,
      ],
      "handleRequestData": [Function],
      "name": "offers",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            aria-current="page"
            className="cta button is-primary"
            to="/offres/nouveau"
>
            <span
                        className="icon"
            >
                        <Icon
                                    svg="ico-offres-w"
                        />
            </span>
            <span>
                        Créer une offre
            </span>
</NavLink>,
          "title": "Vos offres",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": Array [
              <span
                className="icon"
>
                <Icon
                                svg="ico-offres-w"
                />
</span>,
              <span>
                Créer une offre
</span>,
            ],
            "className": "cta button is-primary",
            "to": "/offres/nouveau",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <Icon
                  svg="ico-offres-w"
/>,
                "className": "icon",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "svg": "ico-offres-w",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Créer une offre",
              },
              "ref": null,
              "rendered": "Créer une offre",
              "type": "span",
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label
              className="label"
>
              Rechercher une offre :
</label>,
            <div
              className="field is-grouped"
>
              <p
                            className="control is-expanded"
              >
                            <input
                                          className="input"
                                          defaultValue={undefined}
                                          id="search"
                                          placeholder="Saisissez un ou plusieurs mots complets"
                                          type="text"
                            />
              </p>
              <p
                            className="control"
              >
                            <button
                                          className="button is-primary is-outlined"
                                          id="search-button"
                                          type="submit"
                            >
                                          OK
                            </button>
                             
                            <button
                                          className="button is-secondary"
                                          disabled={true}
                            >
                                           
                                          <Icon
                                                        svg="ico-filter"
                                          />
                                           
                            </button>
              </p>
</div>,
          ],
          "className": "section",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Rechercher une offre :",
              "className": "label",
            },
            "ref": null,
            "rendered": "Rechercher une offre :",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p
                  className="control is-expanded"
>
                  <input
                                    className="input"
                                    defaultValue={undefined}
                                    id="search"
                                    placeholder="Saisissez un ou plusieurs mots complets"
                                    type="text"
                  />
</p>,
                <p
                  className="control"
>
                  <button
                                    className="button is-primary is-outlined"
                                    id="search-button"
                                    type="submit"
                  >
                                    OK
                  </button>
                   
                  <button
                                    className="button is-secondary"
                                    disabled={true}
                  >
                                     
                                    <Icon
                                                      svg="ico-filter"
                                    />
                                     
                  </button>
</p>,
              ],
              "className": "field is-grouped",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <input
                    className="input"
                    defaultValue={undefined}
                    id="search"
                    placeholder="Saisissez un ou plusieurs mots complets"
                    type="text"
/>,
                  "className": "control is-expanded",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "input",
                    "defaultValue": undefined,
                    "id": "search",
                    "placeholder": "Saisissez un ou plusieurs mots complets",
                    "type": "text",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "input",
                },
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <button
                      className="button is-primary is-outlined"
                      id="search-button"
                      type="submit"
>
                      OK
</button>,
                    " ",
                    <button
                      className="button is-secondary"
                      disabled={true}
>
                       
                      <Icon
                                            svg="ico-filter"
                      />
                       
</button>,
                  ],
                  "className": "control",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "OK",
                      "className": "button is-primary is-outlined",
                      "id": "search-button",
                      "type": "submit",
                    },
                    "ref": null,
                    "rendered": "OK",
                    "type": "button",
                  },
                  " ",
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": Array [
                        " ",
                        <Icon
                          svg="ico-filter"
/>,
                        " ",
                      ],
                      "className": "button is-secondary",
                      "disabled": true,
                    },
                    "ref": null,
                    "rendered": Array [
                      " ",
                      Object {
                        "instance": null,
                        "key": undefined,
                        "nodeType": "function",
                        "props": Object {
                          "svg": "ico-filter",
                        },
                        "ref": null,
                        "rendered": null,
                        "type": [Function],
                      },
                      " ",
                    ],
                    "type": "button",
                  },
                ],
                "type": "p",
              },
            ],
            "type": "div",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "section",
        },
        "ref": null,
        "rendered": null,
        "type": "ul",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            false,
            <withRouter(_withQueryRouter)
              className="offers-list main-list"
              element="ul"
              hasMore={false}
              isLoading={true}
              loader={
                            <Spinner
                                          Tag="div"
                                          className={null}
                                          dotFrequency={500}
                                          label="Chargement"
                                          style={null}
                            />
              }
              useWindow={true}
>
              
</withRouter(_withQueryRouter)>,
          ],
          "className": "section",
        },
        "ref": null,
        "rendered": Array [
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [],
              "className": "offers-list main-list",
              "element": "ul",
              "hasMore": false,
              "isLoading": true,
              "loader": <Spinner
                Tag="div"
                className={null}
                dotFrequency={500}
                label="Chargement"
                style={null}
/>,
              "useWindow": true,
            },
            "ref": null,
            "rendered": Array [],
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <HeroSection
            title="Vos offres"
>
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="cta button is-primary"
                        to="/offres/nouveau"
            >
                        <span
                                    className="icon"
                        >
                                    <Icon
                                                svg="ico-offres-w"
                                    />
                        </span>
                        <span>
                                    Créer une offre
                        </span>
            </NavLink>
</HeroSection>,
          <form
            className="section"
            onSubmit={[Function]}
>
            <label
                        className="label"
            >
                        Rechercher une offre :
            </label>
            <div
                        className="field is-grouped"
            >
                        <p
                                    className="control is-expanded"
                        >
                                    <input
                                                className="input"
                                                defaultValue={undefined}
                                                id="search"
                                                placeholder="Saisissez un ou plusieurs mots complets"
                                                type="text"
                                    />
                        </p>
                        <p
                                    className="control"
                        >
                                    <button
                                                className="button is-primary is-outlined"
                                                id="search-button"
                                                type="submit"
                                    >
                                                OK
                                    </button>
                                     
                                    <button
                                                className="button is-secondary"
                                                disabled={true}
                                    >
                                                 
                                                <Icon
                                                            svg="ico-filter"
                                                />
                                                 
                                    </button>
                        </p>
            </div>
</form>,
          <ul
            className="section"
/>,
          <div
            className="section"
>
            <withRouter(_withQueryRouter)
                        className="offers-list main-list"
                        element="ul"
                        hasMore={false}
                        isLoading={true}
                        loader={
                                    <Spinner
                                                Tag="div"
                                                className={null}
                                                dotFrequency={500}
                                                label="Chargement"
                                                style={null}
                                    />
                        }
                        useWindow={true}
            >
                        
            </withRouter(_withQueryRouter)>
</div>,
        ],
        "handleRequestData": [Function],
        "name": "offers",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              aria-current="page"
              className="cta button is-primary"
              to="/offres/nouveau"
>
              <span
                            className="icon"
              >
                            <Icon
                                          svg="ico-offres-w"
                            />
              </span>
              <span>
                            Créer une offre
              </span>
</NavLink>,
            "title": "Vos offres",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": Array [
                <span
                  className="icon"
>
                  <Icon
                                    svg="ico-offres-w"
                  />
</span>,
                <span>
                  Créer une offre
</span>,
              ],
              "className": "cta button is-primary",
              "to": "/offres/nouveau",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Icon
                    svg="ico-offres-w"
/>,
                  "className": "icon",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "svg": "ico-offres-w",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Créer une offre",
                },
                "ref": null,
                "rendered": "Créer une offre",
                "type": "span",
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label
                className="label"
>
                Rechercher une offre :
</label>,
              <div
                className="field is-grouped"
>
                <p
                                className="control is-expanded"
                >
                                <input
                                                className="input"
                                                defaultValue={undefined}
                                                id="search"
                                                placeholder="Saisissez un ou plusieurs mots complets"
                                                type="text"
                                />
                </p>
                <p
                                className="control"
                >
                                <button
                                                className="button is-primary is-outlined"
                                                id="search-button"
                                                type="submit"
                                >
                                                OK
                                </button>
                                 
                                <button
                                                className="button is-secondary"
                                                disabled={true}
                                >
                                                 
                                                <Icon
                                                                svg="ico-filter"
                                                />
                                                 
                                </button>
                </p>
</div>,
            ],
            "className": "section",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Rechercher une offre :",
                "className": "label",
              },
              "ref": null,
              "rendered": "Rechercher une offre :",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <p
                    className="control is-expanded"
>
                    <input
                                        className="input"
                                        defaultValue={undefined}
                                        id="search"
                                        placeholder="Saisissez un ou plusieurs mots complets"
                                        type="text"
                    />
</p>,
                  <p
                    className="control"
>
                    <button
                                        className="button is-primary is-outlined"
                                        id="search-button"
                                        type="submit"
                    >
                                        OK
                    </button>
                     
                    <button
                                        className="button is-secondary"
                                        disabled={true}
                    >
                                         
                                        <Icon
                                                            svg="ico-filter"
                                        />
                                         
                    </button>
</p>,
                ],
                "className": "field is-grouped",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <input
                      className="input"
                      defaultValue={undefined}
                      id="search"
                      placeholder="Saisissez un ou plusieurs mots complets"
                      type="text"
/>,
                    "className": "control is-expanded",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "input",
                      "defaultValue": undefined,
                      "id": "search",
                      "placeholder": "Saisissez un ou plusieurs mots complets",
                      "type": "text",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  "type": "p",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <button
                        className="button is-primary is-outlined"
                        id="search-button"
                        type="submit"
>
                        OK
</button>,
                      " ",
                      <button
                        className="button is-secondary"
                        disabled={true}
>
                         
                        <Icon
                                                svg="ico-filter"
                        />
                         
</button>,
                    ],
                    "className": "control",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "OK",
                        "className": "button is-primary is-outlined",
                        "id": "search-button",
                        "type": "submit",
                      },
                      "ref": null,
                      "rendered": "OK",
                      "type": "button",
                    },
                    " ",
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": Array [
                          " ",
                          <Icon
                            svg="ico-filter"
/>,
                          " ",
                        ],
                        "className": "button is-secondary",
                        "disabled": true,
                      },
                      "ref": null,
                      "rendered": Array [
                        " ",
                        Object {
                          "instance": null,
                          "key": undefined,
                          "nodeType": "function",
                          "props": Object {
                            "svg": "ico-filter",
                          },
                          "ref": null,
                          "rendered": null,
                          "type": [Function],
                        },
                        " ",
                      ],
                      "type": "button",
                    },
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "section",
          },
          "ref": null,
          "rendered": null,
          "type": "ul",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              false,
              <withRouter(_withQueryRouter)
                className="offers-list main-list"
                element="ul"
                hasMore={false}
                isLoading={true}
                loader={
                                <Spinner
                                                Tag="div"
                                                className={null}
                                                dotFrequency={500}
                                                label="Chargement"
                                                style={null}
                                />
                }
                useWindow={true}
>
                
</withRouter(_withQueryRouter)>,
            ],
            "className": "section",
          },
          "ref": null,
          "rendered": Array [
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [],
                "className": "offers-list main-list",
                "element": "ul",
                "hasMore": false,
                "isLoading": true,
                "loader": <Spinner
                  Tag="div"
                  className={null}
                  dotFrequency={500}
                  label="Chargement"
                  style={null}
/>,
                "useWindow": true,
              },
              "ref": null,
              "rendered": Array [],
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
